examples/arkode/CXX_manyvector/ark_sod_lsrk.hpp

```diff
@@ -0,0 +1,512 @@
+#include "arkode/arkode_erkstep.h"
+#include "arkode/arkode_lsrkstep.h"
+#include "nvector/nvector_manyvector.h"
+#include "nvector/nvector_serial.h"
+#include "sundials/sundials_core.hpp"
+
+// Macros for problem constants
+#define rhoL   SUN_RCONST(1.0)
+#define rhoR   SUN_RCONST(0.125)
+#define pL     SUN_RCONST(1.0)
+#define pR     SUN_RCONST(0.1)
+#define uL     SUN_RCONST(0.0)
+#define uR     SUN_RCONST(0.0)
+#define HALF   SUN_RCONST(0.5)
+#define ZERO   SUN_RCONST(0.0)
+#define ONE    SUN_RCONST(1.0)
+#define TWO    SUN_RCONST(2.0)
+#define FOURTH SUN_RCONST(0.25)
+
+#define NSPECIES 5
+
+#define WIDTH (10 + numeric_limits<sunrealtype>::digits10)
+
+using namespace std;
+
+// -----------------------------------------------------------------------------
+// Problem options
+// -----------------------------------------------------------------------------
+
+class ARKODEParameters
+{
+public:
+  // Integration method (ARKODE_LSRK_SSPs_2, ARKODE_LSRK_SSPs_3, ARKODE_LSRK_SSP10_4,
+  // or any valid ARKODE_ERKTableID for ERK methods)
+  string integrator;
+
+  // Method stages (0 => to use the default; ignored if using ARKODE_LSRK_SSP10_4 or
+  // an ERK method)
+  int stages;
+
+  // Relative and absolute tolerances
+  sunrealtype rtol;
+  sunrealtype atol;
+
+  // Step size selection (ZERO = adaptive steps)
+  sunrealtype fixed_h;
+
+  // Maximum number of time steps between outputs
+  int maxsteps;
+
+  // Output-related information
+  int output;    // 0 = none, 1 = stats, 2 = disk, 3 = disk with tstop
+  int nout;      // number of output times
+  ofstream uout; // output file stream
+
+  // constructor (with default values)
+  ARKODEParameters()
+    : integrator("ARKODE_LSRK_SSP10_4"),
+      stages(0),
+      rtol(SUN_RCONST(1.e-4)),
+      atol(SUN_RCONST(1.e-11)),
+      fixed_h(ZERO),
+      maxsteps(10000),
+      output(1),
+      nout(10){};
+
+}; // end ARKODEParameters
+
+// -----------------------------------------------------------------------------
+// Problem parameters
+// -----------------------------------------------------------------------------
+
+// user data class
+class EulerData
+{
+public:
+  ///// domain related data /////
+  long int nx;    // global number of x grid points
+  sunrealtype t0; // time domain extents
+  sunrealtype tf;
+  sunrealtype xl; // spatial domain extents
+  sunrealtype xr;
+  sunrealtype dx; // spatial mesh spacing
+
+  ///// problem-defining data /////
+  sunrealtype gamma; // ratio of specific heat capacities, cp/cv
+
+  ///// reusable arrays for WENO flux calculations /////
+  sunrealtype* flux;
+  sunrealtype w1d[6][NSPECIES];
+
+  ///// class operations /////
+
+  // constructor
+  EulerData()
+    : nx(512),
+      t0(ZERO),
+      tf(SUN_RCONST(0.1)),
+      xl(ZERO),
+      xr(ONE),
+      dx(ZERO),
+      gamma(SUN_RCONST(1.4)),
+      flux(NULL){};
+
+  // manual destructor
+  void FreeData()
+  {
+    if (flux != NULL)
```

Comment:
This check is redundant

Reply:
I mean the if statement can be replaced with
```
delete[] flux;
flux = nullptr;
```
This is equivalent since freeing a null pointer is fine.